src/v/archival/ntp_archiver_service.cc

```diff
@@ -286,6 +280,8 @@ ss::future<> ntp_archiver::upload_topic_manifest() {
 ss::future<> ntp_archiver::upload_until_term_change() {
     ss::lowres_clock::duration backoff = _conf->upload_loop_initial_backoff;
 
+    auto reset_value_metrics = ss::defer(
```

Comment:
It's possible for term to change while the node remains leader: we shouldn't reset in that case.

I think we can get a neater behavior by moving up to upload_until_abort, and doing the reset in the ` if (!_parent.is_elected_leader()) {` branch